### ISSUE TITLE
add_field and field are not behaving the same way

### ISSUE THREAD

**Issue by sayduck-daniel** [rmosolgo/graphql-ruby#2158]
My root query (`QueryType`) became too long and I'm trying to split its fields into several files.

I encounter a problem in transferring a field that uses `.connection_type`. It seems to me the behaviour should be the same for this:

```graphql
query Test {
  fooNew(id: "123") { nodes { id } }
  fooOriginal(id: "123") { nodes { id } }
}
```

but I get this answer:

```json
{
  "data": {
    "fooNew": {
      "nodes": null
    },
    "fooOriginal": {
      "nodes": [
        {
          "id": "123"
        }
      ]
    },
  "errors": [
    {
      "message": "undefined method `edge_nodes' for #<Foo::ActiveRecord_Relation:0x00007fb8dc75a660>",
      "locations": [
        {
          "line": 6,
          "column": 5
        }
      ],
      "path": [
        "fooNew",
        "nodes"
      ]
    }
  ]
```

That is pretty much my code (took of useless parts and logic)

#### graphql/types/query_type.rb
```ruby
# frozen_string_literal: true

module Types
  class QueryType < BaseObject
    add_field(FieldTypes::FooNewField)

    field :foo_original, Types::FooType.connection_type, '...', null: true do
      argument :id, ID, '...', required: true
    end

    def foo_original(id:)
      some_object = MySchema.object_from_id(id, context)
      Foo.where(some_object: some_object)
    end
  end
end
```

#### graphql/fields_types/foo_new_field.rb
```ruby
module FieldTypes
  FooNewField = BaseField.new(
    name: :foo_new,
    type: Types::FooType.connection_type,
    null: true,
    description: '...'
  ) do
    argument :id, GraphQL::Types::ID, '...', required: true

    def resolve(_object, arguments, context)
      some_object = MySchema.object_from_id(arguments[:id], context)
      Foo.where(some_object: some_object)
    end
  end
end
```

#### graphql/field_types/base_field.rb
```ruby
module FieldTypes
  class BaseField < GraphQL::Schema::Field
    def resolve_field(obj, args, ctx)
      resolve(obj, args, ctx)
    end
  end
end
```

#### graphql/types/foo_type.rb
```ruby
module Types
  class FooType < BaseCustomObject
    field :name, String, 'The name of foo', null: true
    field :uuid, String, 'The UUID of foo', null: false
  end
end
```

#### graphql/types/base_custom_object
```ruby
module Types
  class BaseCustomObject < BaseObject
    implements GraphQL::Relay::Node.interface
    connection_type_class ConnectionTypes::BaseConnection
    global_id_field :id
    field :created_at, ScalarTypes::DateType, 'The creation date.', null: false
    field :updated_at, ScalarTypes::DateType, 'The latest update date.', null: false
  end
end
```

#### graphql/types/base_object
```ruby
module Types
  class BaseObject < GraphQL::Schema::Object
  end
end
```

#### graphql/conection_types/base_connection.rb
```ruby
module ConnectionTypes
  class BaseConnection < GraphQL::Types::Relay::BaseConnection
    field :total_count, Integer, 'Amount of records for this connection', null: false
    def total_count
      object.nodes.size
    end

    field :count, Integer, 'Amount of records for this edge', null: false
    def count
      object.edge_nodes.count
    end
  end
end
```

Any idea what this could be or where to start looking?

**Comment by rmosolgo** [comment]
Hey, it seems like it's not getting automatically wrapped with a `RelationConnection`. I'm not sure why but you could try adding `BaseField.new(... connection: true ...)` and see if that helps it. 

There was just another issue with `add_field(...)` (#2153), it's obviously not doing a great job 😬

**Comment by sayduck-daniel** [comment]
Just tried, still the same error

**Comment by sayduck-daniel** [comment]
Just to be sure, what I did:

```ruby
module FieldTypes
  FooNewField = BaseField.new(
    name: :foo_new,
    type: Types::FooType.connection_type,
    null: true,
    description: '...',
    connection: true # This line was added
  ) do
    # ...
  end
end
```

**Comment by sayduck-daniel** [comment]
Still blocked on this issue. 

Any other idea how I could use a field with a Connection type?

**Comment by rmosolgo** [comment]
Sorry, I'm not sure what else to try. I'd recommend refactoring to resolvers instead: https://graphql-ruby.org/fields/resolvers.html

Unlike `add_field`, Schema::Resolver is a well-tested public API.

**Comment by sayduck-daniel** [comment]
FYI This is still happening with `GraphQL::Execution::Interpreter`

**Comment by rmosolgo** [comment]
I don't have plans to support `add_field` except for `Types::Relay::NodeField`, so I'll close this issue. If you give it a try with a resolver and run into trouble, please open a new issue!
